homeassistant/components/comfoair/sensor.py

```diff
@@ -0,0 +1,158 @@
+"""Platform to control a Zehnder ComfoAir 350 ventilation unit."""
+
+import logging
+
+from homeassistant.const import CONF_RESOURCES, TEMP_CELSIUS
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from . import DOMAIN, ComfoAirModule, SIGNAL_COMFOAIR_UPDATE_RECEIVED
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_COMFORT_TEMPERATURE = "comfort_temperature"
+ATTR_OUTSIDE_TEMPERATURE = "outside_temperature"
+ATTR_SUPPLY_TEMPERATURE = "supply_temperature"
+ATTR_RETURN_TEMPERATURE = "return_temperature"
+ATTR_EXHAUST_TEMPERATURE = "exhaust_temperature"
+ATTR_AIR_FLOW_SUPPLY = "air_flow_supply"
+ATTR_AIR_FLOW_EXHAUST = "air_flow_exhaust"
+ATTR_FAN_SPEED_MODE = "speed_mode"
+
+SENSOR_TYPES = {
+    ATTR_COMFORT_TEMPERATURE: [
+        "Comfort Temperature",
+        TEMP_CELSIUS,
+        "mdi:thermometer",
+        0xD2,
+        0 * 8,
+        8,
+    ],
+    ATTR_OUTSIDE_TEMPERATURE: [
+        "Outside Temperature",
+        TEMP_CELSIUS,
+        "mdi:thermometer",
+        0xD2,
+        1 * 8,
+        8,
+    ],
+    ATTR_SUPPLY_TEMPERATURE: [
+        "Supply Temperature",
+        TEMP_CELSIUS,
+        "mdi:thermometer",
+        0xD2,
+        2 * 8,
+        8,
+    ],
+    ATTR_RETURN_TEMPERATURE: [
+        "Return Temperature",
+        TEMP_CELSIUS,
+        "mdi:thermometer",
+        0xD2,
+        3 * 8,
+        8,
+    ],
+    ATTR_EXHAUST_TEMPERATURE: [
+        "Exhaust Temperature",
+        TEMP_CELSIUS,
+        "mdi:thermometer",
+        0xD2,
+        4 * 8,
+        8,
+    ],
+    ATTR_AIR_FLOW_EXHAUST: ["Exhaust airflow", "%", "mdi:fan", 0xCE, 6 * 8, 8],
+    ATTR_AIR_FLOW_SUPPLY: ["Supply airflow", "%", "mdi:fan", 0xCE, 7 * 8, 8],
+    ATTR_FAN_SPEED_MODE: ["Speed mode", "", "mdi:fan", 0xCE, 8 * 8, 8],
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the ComfoAir sensor platform."""
+    unit = hass.data[DOMAIN]
+
+    sensors = []
+    for resource in config.get(CONF_RESOURCES, []):
+        sensor_type = resource.lower()
+        _LOGGER.debug("Sensor type: %s", sensor_type)
+
+        if sensor_type not in SENSOR_TYPES:
+            _LOGGER.warning("Sensor type: %s is not a valid sensor.", sensor_type)
```

Comment:
Logging messages should not end with a period

Reply:
Actually, you should use vol to validate the sensors config. I recently did the same for comfoconnect, see https://github.com/home-assistant/home-assistant/commit/3d5b007c6bc63be50a4aa5bb8e45e9366bdbd201

Reply:
I'll remove the data selector as recommended by Martin in https://github.com/home-assistant/home-assistant/pull/27961#discussion_r350895455